stdlib/event/gui.lua

```diff
@@ -46,44 +19,144 @@ function Gui.dispatch(event)
 
     if event.element and event.element.valid then
         event.tick = event.tick or game.tick
-        for gui_element_pattern, handler in pairs(Event._registry[event.name]) do
+        for gui_element_pattern, handlers in pairs(Gui._registry[event.name]) do
             if event.element and event.element.valid then
                 local match_str = event.element.name:match(gui_element_pattern)
                 if match_str then
                     event.match = match_str
                     event.state = event.name == defines.events.on_gui_checked_state_changed and event.element.state or nil
                     event.text = event.name == defines.events.on_gui_text_changed and event.element.text or nil
-                    setmetatable(event, { __index = { _handler = handler } })
-                    local success, err = pcall(handler, event)
-                    if not success then
-                        game.print(err)
+                    local metahandler = {}
+                    setmetatable(event, {__index = metahandler})
+                    for _, handler in ipairs(handlers) do
+                        metahandler._handler = handler
+                        local success, err = pcall(handler, event)
+                        if not success then
+                            if game and game.print then
+                                game.print(err)
+                            elseif log then
```

Comment:
log will always be available, it is a protected function

Reply:
ACK, deparanoiafied downstream.